ingestion/functions/parsing/common/python/parsing_lib/parsing_lib.py

```diff
@@ -47,6 +49,9 @@ def write_to_server(cases, headers):
     res = requests.post(put_api_url, json={"cases": cases},
                        headers=headers)
     res_json = res.json()
+    if res.status_code != 200:
+        raise RuntimeError(f'Error sending cases to server, status={res.status_code}, response={res_json}')
```

Comment:
For a later PR to be sure, but we should consolidate the error handling to a helper method. I imagine we'll need to do some writes to the Upload API (soon(tm)) as part of error handling.

Reply:
yup, definitely, there is the option of calling raise_for_status() when doing the post which will raise inc ase of an error status, here I chose to check for the status myself, we could have the upload api path handled in the exception handling code or here in a helper function, we'll see when that time comes I guess.